js/forum_italia.js

```diff
@@ -0,0 +1,295 @@
+      commentBoxesSelector: '.box-comment.can-post',
+      commentBoxesTopicClosedSelector: '.box-comment:not(.can-post)',
+      forumLoginMarkup: tpl({}, 'forum__login'),
+      forumCommentMarkup: tpl({}, 'forum__comment'),
+      suggestionsMarkup: tpl({}, 'forum-italia__markup__tooltip'),
+      logoutModal: tpl({}, 'forum__logout--modal'),
+      userAvatarClass: '.box-comment__figure',
+      publicUserFieldIndex: '2',
+      client: null
+    },
+
+    init: function() {
+      that = this.$;
+
+      if (!that.$forumItaliaComments.length) {
+        return;
+      }
+
+      that.client = new Discourse({
+        appName: 'Docs Italia',
+        apiBaseUrl: 'https://forum.italia.it',
+        scopes: ['write']
+      });
+
+      that.client.init().then(function() {
+        var forumItaliaComments = that.$forumItaliaComments.toArray().map(function(forumItaliaComment) {
+          var $forumItaliaComment = $(forumItaliaComment);
+          var topicId = $forumItaliaComment.data('topic');
+          var postsCount;
+
+          return that.client.getTopic(topicId, true).then(function(topic) {
+            var posts = topic.post_stream.posts;
+
+            $forumItaliaComment.attr('id', 'forum-italia-comments-' + topicId);
+            $forumItaliaComment.html(tpl({
+              topicId: topicId,
+              commentsCount: topic.posts_count,
+              canPost: (!topic.archived && !topic.closed) ? 'can-post' : ''
+            }, 'forum-italia-comments'))
+
+            forumItalia.commentsMarkup($forumItaliaComment.find(that.topicCommentsSelector), posts);
+            
+            return {
+              $element: $forumItaliaComment,
+              topic: topic,
+            }
+          }).catch(function(error) { /* topic not found or not public */ });
+        });
+
+        Promise.all(forumItaliaComments).then(function(forumItaliaCommentsTopics) {
+          forumItaliaCommentsTopics.map(function(forumItaliaCommentsTopic) {
+            forumItalia.headingMarkup(forumItaliaCommentsTopic.$element.closest('.section'), forumItaliaCommentsTopic.topic.id, forumItaliaCommentsTopic.topic.posts_count);
+          });
+          that.client.isLoggedIn().then(function(isLoggedIn) {
+            if (isLoggedIn) {
+              forumItalia.setLoggedInMarkup();
+            } else {
+              forumItalia.setLoggedOutMarkup();
+            }
+            forumItalia.setClosedTopicsMarkup();
+          });
+        });
+      });
+    },
+
+    headingMarkup: function($section, topicId, postsCount) {
+      $section.find('.chapter-nav__list.chapter-nav__list--visible').first().children('.chapter-nav__item').last().before(tpl({
+        postsCount: postsCount,
+        topicId: topicId
+      }, 'section_navigation__comments'));
+    },
+
+    login: function() {
+      that.client.login().then(function() {
+        forumItalia.setLoggedInMarkup();
+      }).catch(function(error) {
+        console.log('error', error);
```

Comment:
Userei per tutti i `catch` in questo file una gestione univoca, in modo da poterla nel caso sostituire con più semplicità in futuro.